src/io_uring.c

```diff
@@ -0,0 +1,97 @@
+#include "io_uring.h"
+
+#ifdef HAVE_LIBURING
+#include <liburing.h>
+#include <string.h>
+
+#include "serverassert.h"
+#include "zmalloc.h"
+
+#define IO_URING_DEPTH 256 /* io_uring instance queue depth. */
+
+static size_t io_uring_queue_len = 0;
+static io_uring *_io_uring;
+
+void initIOUring(void) {
+    struct io_uring_params params;
+    struct io_uring *ring = zmalloc(sizeof(struct io_uring));
+    memset(&params, 0, sizeof(params));
+    /* On success, io_uring_queue_init_params(3) returns 0 and ring will
+     * point to the shared memory containing the io_uring queues.
+     * On failure -errno is returned. */
+    assert(io_uring_queue_init_params(IO_URING_DEPTH, ring, &params) == 0);
```

Comment:
let's use serverAssert. `assert` is mapped to `serverAssert` anyways but IMO adds more confusion as `assert` is already a well established c construct. 

```suggestion
    serverAssert(io_uring_queue_init_params(IO_URING_DEPTH, ring, &params) == 0);
```

Reply:
`serverAssert` is defined in `server.h` now, we don't want io_uring have dependency of `server.h`. 
io_uring.c include `serverassert.h`, the `assert` is same with `serverAssert`. 
I saw `sds.c` `dict.c` used `assert` directly, maybe we can keep alignment with them?

Reply:
This is not a blocker to me but I will probably clean this up some other time (for consistency) :-)